testing/scenario_app/ios/Scenarios/ScenariosUITests/TextSemanticsFocusTest.m

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "TextSemanticsFocusTest.h"
+
+@implementation TextSemanticsFocusTest
+
+- (void)setUp {
+  [super setUp];
+  self.continueAfterFailure = NO;
+
+  self.application = [[XCUIApplication alloc] init];
+  self.application.launchArguments = @[ @"--text-semantics-focus" ];
+  [self.application launch];
+}
+
+- (void)testAccessibilityFocusOnTextSemanticsProducesCorrectIosViews {
+  // Find the initial TextInputSemanticsObject which was sent from the mock framework on first
+  // frame.
+  XCUIElement* textInputSemanticsObject =
+      [[[self.application textFields] matchingIdentifier:@"flutter textfield"] element];
+  XCTAssertTrue([textInputSemanticsObject waitForExistenceWithTimeout:5]);
+  XCTAssertEqual([textInputSemanticsObject valueForKey:@"hasKeyboardFocus"], @(NO));
```

Comment:
XCTAssertFalse, you are doing pointer comparisons on objects.

Reply:
FYI alternatively you could use, xctassertequalobjects.

Reply:
I think `@NO`/`__objc_no` is a singleton and is a true-y (!= nil) value 🤣
But XCTAssertEqualObjects does show a nicer error when it doesn't match. Changed.